numpy/core/numeric.py

```diff
@@ -141,8 +141,13 @@ def zeros_like(a, dtype=None, order='K', subok=True, shape=None):
     return res
 
 
+def _ones_dispatcher(shape, dtype=None, order=None, like=None):
+    return(like,)
+
+
+@array_function_dispatch(_ones_dispatcher)
 @set_module('numpy')
-def ones(shape, dtype=None, order='C'):
+def ones(shape, dtype=None, order='C', like=None):
```

Comment:
If `like` is only checked in `**kwargs`, then we should probably also require that it's called as a keyword argument, e.g.,
```suggestion
def ones(shape, dtype=None, order='C', *, like=None):
```

Reply:
Yes, that makes sense. Do you think we should do it in all function signatures, and perhaps even amend NEP-35 to reflect that?

Reply:
Or maybe amending NEP-35 would not be necessary as this would be more of an implementation detail only. I don't have strong feelings either way.

Reply:
The alternative would looking for a `like` argument explicitly by explicitly parsing the arguments in each function call. This is possible but seems much more error prone.

I think adding a few sentences to clarify this in the NEP would not be a bad idea. The argument parsing is much simpler / more uniform, and even if we didn't care about the implementation complexity, it already is better API design to make most optional keyword arguments keyword-only. NumPy doesn't do much of this currently only because it's a Python 3 only feature.

Reply:
I think it'd be worthwhile to make it a keyword-only argument.

Reply:
Thanks both for the feedback, I just pushed a commit to make `like=` a keyword-only argument everywhere. I'll also start a new PR to update NEP-35 as discussion here progresses.